components/brave_wallet_ui/common/hardware/ledgerjs/ledger-untrusted-transport.ts

```diff
@@ -31,16 +32,25 @@ export class LedgerUntrustedMessagingTransport extends LedgerMessagingTransport
     this.addCommandHandler<SignTransactionResponsePayload>(LedgerCommand.SignTransaction, this.handleSignTransaction)
   }
 
-  handleUnlock = async (commandx: UnlockCommand): Promise<UnlockResponsePayload> => {
-    if ((await this.authorizationNeeded())) { // TODO should try / catch this, possible a device error occurs
-      const error = {
+  handleUnlock = async (command: UnlockCommand): Promise<UnlockResponsePayload> => {
+    let payload: LedgerResponse | HardwareOperationResult
+    if ((await this.authorizationNeeded())) {
+      payload = {
         success: false,
         error: 'unauthorized',
         code: 'unauthorized'
       }
-      return this.createUnlockResponse(commandx, false, error)
+    } else {
+      payload = { success: true }
     }
-    return this.createUnlockResponse(commandx, true)
+
+    const responsePayload: UnlockResponsePayload = {
+      id: command.id,
+      command: command.command,
+      payload: payload,
```

Comment:
Odd seeing types with 'Payload' in the name that also includes a 'payload' property. Shouldn't the type name be "UnlockResponse"?